### PR TITLE
Test that types returned by database match the query model

### PR DESCRIPTION
**Pull request by Jongmassey**
Addresses #706

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -109,3 +109,6 @@ story-dependencies.png
 # Build/test byproducts
 cache/
 .hypothesis
+
+# VS Code configuration
```
Comment (Jongmassey) [inline .gitignore:113]: Sorry if this is a bit of a drive-by. Surely I can't be the only one using vs code?

Comment (evansd) [reply .gitignore:113]: You can set this as part of the global gitignore for your system. In general, detritus created by the project should be excluded by the project gitignore, and detritus created by the user's tooling/operating system etc should be excluded by the user's gitignore.

This should be just be as simple as creating a file in `~/.config/git/ignore`, but there are some more details here if that's helpful:
https://git-scm.com/docs/gitignore

Comment (Jongmassey) [reply .gitignore:113]: Ideal, shall do thanks!

File: tests/spec/conftest.py [hunk 2]
```diff
@@ -41,6 +43,11 @@ def run_test_execute(table_data, series, expected_results, population=None):
         assert len(results) == len(results_dict), "Duplicate patient IDs found"
         assert results_dict == expected_results
 
+        # assert types are as expected
+        variables = compile(dataset)
+        variable_type = get_series_type(variables["v"])
+        assert results[0][1] is None or isinstance(results[0][1], variable_type)
```
Comment (evansd) [inline tests/spec/conftest.py:49]: This only checks the first result, which is not unreasonable as they should all have the same type unless something _really_ horrible has happened. But, if the first value happens to be `None` then it's not going to do any type-checking at all.

You could make it check the first non-null value, but it's probably simpler to check the type of every value than it is to do that.

**Review by evansd — APPROVED** [review]
Happy to approve this as I think in practice it enforces what we need, but have added a note about what happens if the first value is `None`.

File: tests/spec/conftest.py (hunk 2 above)
Comment (Jongmassey) [reply tests/spec/conftest.py]: Addressed in latest commit (I hope!)

**Review by evansd — APPROVED** [review]
Great, thanks!

**Pull request merged**